opentech/apply/funds/blocks.py

```diff
@@ -1,9 +1,129 @@
+from collections import Counter
+
+from django import forms
+from django.core.exceptions import ValidationError
+from django.forms.utils import ErrorList
 from django.utils.translation import ugettext_lazy as _
+from django.utils.text import mark_safe
 
-from opentech.apply.stream_forms.blocks import FormFieldsBlock
+from wagtail.wagtailcore.blocks import StaticBlock
+from opentech.apply.stream_forms.blocks import FormFieldsBlock, FormFieldBlock
 
 from opentech.apply.categories.blocks import CategoryQuestionBlock
 
 
+def find_duplciates(items):
+    counted = Counter(items)
+    duplicates = [
+        name for name, count in counted.items() if count > 1
+    ]
+    return duplicates
+
+
 class CustomFormFieldsBlock(FormFieldsBlock):
     category = CategoryQuestionBlock(group=_('Custom'))
+
+    def __init__(self, *args, **kwargs):
+        child_blocks = [(block.name, block(group=_('Required'))) for block in MustIncludeFieldBlock.__subclasses__()]
+        super().__init__(child_blocks, *args, **kwargs)
+
+    def clean(self, value):
+        try:
+            value = super().clean(value)
+        except ValidationError as e:
+            error_dict = e.params
+        else:
+            error_dict = dict()
+
+        block_types = [block.block_type for block in value]
+        missing = set(REQUIRED_BLOCK_NAMES) - set(block_types)
+
+        duplicates = [
+            name for name in find_duplciates(block_types)
+            if name in REQUIRED_BLOCK_NAMES
+        ]
+
+        all_errors = list()
+        if missing:
+            all_errors.append(
+                'You are missing the following required fields: {}'.format(', '.join(missing).title())
+            )
+
+        if duplicates:
+            all_errors.append(
+                'You have duplicates of the following required fields: {}'.format(', '.join(duplicates).title())
+            )
+            for i, block_name in enumerate(block_types):
+                if block_name in duplicates:
+                    self.add_error_to_child(error_dict, i, 'info', 'Duplicate field')
+
+        if all_errors or error_dict:
+            error_dict['__all__'] = all_errors
+            raise ValidationError('Error', params=error_dict)
+
+        return value
+
+    def add_error_to_child(self, errors, child_number, field, message):
```

Comment:
PyCharm inspections suggest it is a static method

Reply:
It is, but what value does defining it as static give us?

Reply:
A less shouty IDE. At the same time, it doesn't give us anything, so meh..